### PR TITLE
[10.0][MIG] Migrates web_domain_field module

### PR DESCRIPTION
**Pull request by rousseldenis**

### CODE REVIEW

File: web_domain_field/static/src/js/pyeval.js [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+//value as domain
+function eval_domains (domains, evaluation_context) {
+    evaluation_context = _.extend(pyeval.context(), evaluation_context ||
+    {});
+    var result_domain = [];
+    _(domains).each(function (domain) {
+        if (_.isString(domain)) {
+           // Modified part or the original method
+           if(domain in evaluation_context) {
+                result_domain.push.apply(
```
Comment (grindtildeath) [inline web_domain_field/static/src/js/pyeval.js]: Hi @rousseldenis, @lmignon ,
I tried this port of v9 module but I get an error on this apply function here.
Don't know what could I do to help, so here is at least the error I get and my example code.

```
TypeError: second argument to Function.prototype.apply must be an array

http://localhost:8069/web/content/307-c8e6145/web.assets_backend.js:4267
Traceback:
eval_domains/<@http://localhost:8069/web/content/307-c8e6145/web.assets_backend.js:4267:230
_.forEach@http://localhost:8069/web/content/293-31cc17f/web.assets_common.js:12:558
_.mixin/</_.prototype[name]@http://localhost:8069/web/content/293-31cc17f/web.assets_common.js:69:521
eval_domains@http://localhost:8069/web/content/307-c8e6145/web.assets_backend.js:4267:141
eval_domains/<@http://localhost:8069/web/content/307-c8e6145/web.assets_backend.js:4269:301
_.forEach@http://localhost:8069/web/content/293-31cc17f/web.assets_common.js:12:558
_.mixin/</_.prototype[name]@http://localhost:8069/web/content/293-31cc17f/web.assets_common.js:69:521
eval_domains@http://localhost:8069/web/content/307-c8e6145/web.assets_backend.js:4267:141
eval_domains/<@http://localhost:8069/web/content/307-c8e6145/web.assets_backend.js:4269:301
_.forEach@http://localhost:8069/web/content/293-31cc17f/web.assets_common.js:12:558
_.mixin/</_.prototype[name]@http://localhost:8069/web/content/293-31cc17f/web.assets_common.js:69:521
eval_domains@http://localhost:8069/web/content/307-c8e6145/web.assets_backend.js:4267:141
domain_field_pyeval@http://localhost:8069/web/content/307-c8e6145/web.assets_backend.js:4271:24
eval_arg@http://localhost:8069/web/content/307-c8e6145/web.assets_backend.js:4266:61
ensure_evaluated@http://localhost:8069/web/content/307-c8e6145/web.assets_backend.js:4260:11
call@http://localhost:8069/web/content/307-c8e6145/web.assets_backend.js:1476:1
name_search@http://localhost:8069/web/content/307-c8e6145/web.assets_backend.js:1758:1038
get_search_result@http://localhost:8069/web/content/307-c8e6145/web.assets_backend.js:2224:25
source@http://localhost:8069/web/content/307-c8e6145/web.assets_backend.js:2332:103
_search@http://localhost:8069/web/content/293-31cc17f/web.assets_common.js:1461:142
$.widget/</proxiedPrototype[prop]</<@http://localhost:8069/web/content/293-31cc17f/web.assets_common.js:1021:328
search@http://localhost:8069/web/content/293-31cc17f/web.assets_common.js:1461:8
$.widget/</proxiedPrototype[prop]</<@http://localhost:8069/web/content/293-31cc17f/web.assets_common.js:1021:328
$.widget.bridge/$.fn[name]/<@http://localhost:8069/web/content/293-31cc17f/web.assets_common.js:1026:13
each@http://localhost:8069/web/content/293-31cc17f/web.assets_common.js:370:758
each@http://localhost:8069/web/content/293-31cc17f/web.assets_common.js:358:636
$.widget.bridge/$.fn[name]@http://localhost:8069/web/content/293-31cc17f/web.assets_common.js:1023:248
render_editable/<@http://localhost:8069/web/content/307-c8e6145/web.assets_backend.js:2327:1041
dispatch@http://localhost:8069/web/content/293-31cc17f/web.assets_common.js:637:378
add/elemData.handle@http://localhost:8069/web/content/293-31cc17f/web.assets_common.js:610:151
```

My code looks like this :
```
original_order_products_domain = fields.Char(
        'product.product', compute="_compute_original_order_products_domain")
    original_order_id = fields.Many2one(
        'sale.order',
        string='Original sale order'
    )

    @api.multi
    @api.depends('original_order_id')
    def _compute_original_order_products_domain(self):
        for r in self:
            order_lines = self.env['sale.order.line'].search(
                [('order_id', '=', r.original_order_id.id)])
            if order_lines:
                product_ids = order_lines.mapped('product_id').ids
                r.original_order_products_domain = json.dumps(
                    "[('id', 'in', %s)]" % product_ids)
```
And on the view :
```
<field name="original_order_products_domain" readonly="1"/>
<field name="product_id" domain="original_order_products_domain"/>
```

**Review by grindtildeath — APPROVED** [review]
Sorry for first comment, I did dump a string in json but it works well !

**Review by adrienpeiffer — APPROVED** [review]
:+1: (Tested)

**Comment by rousseldenis** [comment]
@pedrobaeza Can we move forward on this ?

**Comment by pedrobaeza** [comment]
Are you sure this is still needed? There was a commit a couple of months ago that standarizes the values stored on JS for x2many, making the domain construction easier.

**Comment by rousseldenis** [comment]
@pedrobaeza Can you give me a pointer ?

**Comment by pedrobaeza** [comment]
I'm not able to find right now the commit. I was proposing to simply try now on a fresh v10 a domain like:
```xml
<field name="field" domain="[('id', 'in', x2many_field)]">
```
to see if it works.

**Comment by rousseldenis** [comment]
@pedrobaeza Ok, but that's not the only case this addon will fix... See README

**Comment by pedrobaeza** [comment]
Yeah, but if it's covered, then you only need a compute many2many for handling the rest of the cases (the same as you do here).

**Comment by lmignon** [comment]
@pedrobaeza But to compute your many2many you must resolve the domain to get the value of the many2many even if you don't need it. Moreover, your many2many could contains thousands of ids which is not optimal in terms of performance....

**Comment by pedrobaeza** [comment]
OK, let's continue then with the addon. I miss the performance reason in the README though. I think it's also a good reason for being mentioned.

**Comment by rousseldenis** [comment]
@pedrobaeza README changed

**Comment by pedrobaeza** [comment]
Thanks, merging

**Pull request merged**